src/Commands/PluginCommand.php

```diff
@@ -0,0 +1,568 @@
+                                'name'        => $name,
+                                'location'    => $location,
+                                'method'      => $method,
+                                'version'     => $version,
+                                'description' => $description,
+                            ];
+                    }
+                }
+            }
+
+            if (empty($rows)) {
+                $this->log()->notice('You have no plugins installed.');
+                return false;
+            }
+
+            $count = count($rows);
+            $plural = ($count > 1) ? 's' : '';
+            $message = "You have {$count} plugin{$plural} installed.  Use 'terminus plugin:install <project>...' to add more plugins.";
+            $this->log()->notice($message);
+
+            // Output the plugin list in table format.
+            return new RowsOfFields($rows);
+        }
+    }
+
+    /**
+     * Search for available Terminus plugins.
+     *
+     * @command plugin:search
+     * @aliases plugin:find plugin:locate
+     *
+     * @option keyword A search string used to query for plugins. Example: terminus plugin:search pantheon.
+     */
+    public function search($keyword = '')
+    {
+        if (empty($keyword)) {
+            $message = "Usage: terminus plugin:<search|find|locate> <string>";
+            $this->log()->error($message);
+            return false;
+        }
+
+        if ($this->commandExists('composer')) {
+            exec("composer search -t terminus-plugin {$keyword}", $output);
+            foreach ($output as $message) {
+                if (stripos($message, 'terminus') && stripos($message, 'plugin')) {
+                    $this->log()->notice($message);
+                }
+            }
+        } else {
+            $message = "In order to search for plugins, you need to install composer.";
+            $this->log()->notice($message);
+        }
+    }
+
+    /**
+     * Update one or more Terminus plugins.
+     *
+     * @command plugin:update
+     * @aliases plugin:upgrade plugin:up
+     *
+     * @option plugin_list A comma delimited list of one or more installed plugins to update
+     *
+     * @usage <plugin-name-1|all>,[plugin-name-2],...
+     */
+    public function update($plugin_list = '')
+    {
+        if (empty($plugin_list)) {
+            $plugin_list = 'all';
+        }
+
+        $plugins = explode(',', $plugin_list);
+        if ($plugins[0] == 'all') {
+            $plugins_dir = $this->getPluginDir();
+            exec("ls \"$plugins_dir\"", $output);
+            if (empty($output[0])) {
+                $message = "You have no plugins installed.";
+                $this->log()->notice($message);
+            } else {
+                foreach ($output as $plugin) {
+                    $this->updatePlugin($plugin);
+                }
+            }
+        } else {
+            foreach ($plugins as $plugin) {
```

Comment:
You don't need to pay much attention to my crazy-old-man code complexity suggestions. I blame Scrutinizer's code quality reports. See [else considered harmful](https://speakerdeck.com/nb/else-considered-harmful) for some interesting perspectives on refactoring.